src/main/scala/org/antipathy/mvn_scalafmt/builder/ChangedFilesBuilder.scala

```diff
@@ -0,0 +1,63 @@
+package org.antipathy.mvn_scalafmt.builder
+
+import java.io.File
+import org.apache.maven.plugin.logging.Log
+import scala.util.control.NonFatal
+import scala.sys.process.ProcessLogger
+
+/**
+  * Class for building a list of files that have changed from a specified git branch
+  * @param log The maven logger
+  * @param diff Should only changed files be returned
+  * @param branch the git branch to compare against
+  * @param changeFunction Function to identify changed files
+  */
+class ChangedFilesBuilder(log: Log, diff: Boolean, branch: String, changeFunction: () => String)
+    extends Builder[Seq[File], Seq[File]] {
+
+  /**
+    * Build a list of files that have changed in git from the specified input
+    *
+    * @param input The input to build from
+    * @return A list of changed files.
+    */
+  override def build(input: Seq[File]): Seq[File] =
+    if (diff) {
+      log.info(s"Checking for files changed from $branch")
+      try {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using util.Try instead of a try-catch block](https://app.codacy.com/manual/Antipathy_org/mvn_scalafmt/pullRequest?prid=4846372)